lib/urlwatch/config.py

```diff
@@ -95,17 +96,27 @@ def parse_args(self, cmdline_args):
         group.add_argument('--test-diff-filter', metavar='JOB',
                            help='test diff filter output of job by location or index (needs at least 2 snapshots)')
         group.add_argument('--dump-history', metavar='JOB', help='dump historical cached data for a job')
+
         group = parser.add_argument_group('interactive commands ($EDITOR/$VISUAL)')
         group.add_argument('--edit', action='store_true', help='edit URL/job list')
         group.add_argument('--edit-config', action='store_true', help='edit configuration file')
         group.add_argument('--edit-hooks', action='store_true', help='edit hooks script')
+
         group = parser.add_argument_group('miscellaneous')
         group.add_argument('--features', action='store_true', help='list supported jobs/filters/reporters')
         group.add_argument('--gc-cache', metavar='RETAIN_LIMIT', type=int, help='remove old cache entries, keeping the latest RETAIN_LIMIT (default: 1)',
                            nargs='?', const=1)
 
         args = parser.parse_args(cmdline_args)
 
-        for i, arg in enumerate(vars(args)):
+        if args.tags:
+            self.tag_set = frozenset(args.joblist)
```

Comment:
Should we warn/fail if `args.joblist` is empty? Not specifying jobs means "all jobs", but just specifying `--tags` without any jobs might warrant a failure?

Reply:
As docced (in this pr) we say specifying nothing means run all jobs, so I'm tempted to go with that behaviour.

Although from a Unix clean interface standpoint, I kind of prefer that it just doesn't run anything and "no tags" isn't a special case. The less special cases the less surprises.

I think the "run all" behaviour is easier to explain (document).

Reply:
I added a test for the behaviour in the base commit, and a new commit on top to switch to running all for the empty list.

If you still think the behaviour should be to error instead, just shout and I'll switch it to that.

Reply:
> If you still think the behaviour should be to error instead, just shout and I'll switch it to that.

Yes, I think failing loudly is good here, otherwise people will have things like `urlwatch --tags $SOME_VARIABLE` and if `$SOME_VARIABLE` is empty (e.g. mistyped or empty for other reasons), then this will not do what the user expects.

What would be the use case of doing nothing at all when running `urlwatch --tags`?

Reply:
The only unix style reason to not error to I can think of is:
```
./tags-to-run.sh | xargs urlwatch --tags
```

And even that can be solved with `xargs --no-run-if-empty`

Anyway, I'm splitting hairs, erroring on an empty tag list is perfectly acceptable behaviour. Have done that.